mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php

```diff
@@ -30,11 +30,22 @@ public function createWorkflow(Workflow $workflow): int {
     if (!$result) {
       throw Exceptions::databaseError($this->wpdb->last_error);
     }
-    $id = (int)$this->wpdb->insert_id;
+    $id = $this->wpdb->insert_id;
     $this->insertWorkflowVersion($id, $workflow);
     return $id;
   }
 
+  public function duplicateWorkflow(Workflow $workflow): int {
+    $data = $workflow->toArray();
+    $now = (new DateTimeImmutable())->format(DateTimeImmutable::W3C);
+    $data['created_at'] = $now;
+    $data['updated_at'] = $now;
+    $data['status'] = Workflow::STATUS_DRAFT;
+    $data['name'] = 'Copy of ' . $workflow->getName();
```

Comment:
This has the potential to cause an error for workflows with names close to the 255 limit

Reply:
Should we increase the limit? Or truncate? I think 255 is fine but not sure if it's bytes or UTF8 characters...

Reply:
I think 255 is fine as well, and is likely much more than anyone will ever use. I'm still going to truncate to 255 characters though, because I don't want to leave the possibility of an API error here that we could easily prevent.